src/battle/battle_system.c

```diff
@@ -1303,3 +1331,634 @@ u32 CalcMoneyLoss(Party *party, PlayerProfile *profile) {
+            Pokedex_SetMonSeenFlag(bsys->pokedex, mon);
+        }
+    }
+    if (!(flag & 1) && GetMonData(mon, MON_DATA_SPECIES_OR_EGG, NULL) == SPECIES_BURMY) {
+        Pokedex_SetMonCaughtFlag(bsys->pokedex, mon);
+    }
+}
+
+void BattleSystem_SetPokedexCaught(BattleSystem *bsys, int battlerId) {
+    u32 flag = ov12_02261258(bsys->opponentData[battlerId]);
+    
+    if (!(bsys->battleType & (BATTLE_TYPE_LINK | BATTLE_TYPE_TOWER)) && (flag & 1)) {
+        int selectedMonIndex = ov12_022581D4(bsys, bsys->ctx, 2, battlerId);
+        Pokemon *mon = BattleSystem_GetPartyMon(bsys, battlerId, selectedMonIndex);
+        Pokedex_SetMonCaughtFlag(bsys->pokedex, mon);
+    }
+}
+
+BOOL BattleSystem_CheckMonCaught(BattleSystem *bsys, int battlerId) {
+    return Pokedex_CheckMonCaughtFlag(bsys->pokedex, battlerId);
+}
+
+void BattleSystem_SetDefaultBlend() {
+    G2_BlendNone();
+}
+
+u8 BattleSystem_PrintTrainerMessage(BattleSystem *bsys, int trainerId, int battlerId, int a2, int delay) {
+    Window *window = BattleSystem_GetWindow(bsys, 0);
+    int index;
+    
+    if (bsys->battleType & BATTLE_TYPE_TOWER) {
+        if (trainerId == 0x2710 || bsys->battleType & BATTLE_TYPE_13) {
+            String *msg;
+
+            if (a2 == 0x64) {
+                msg = MailMsg_GetExpandedString(&bsys->trainers[battlerId].winMessage, HEAP_ID_BATTLE);
+            } else {
+                msg = MailMsg_GetExpandedString(&bsys->trainers[battlerId].loseMessage, HEAP_ID_BATTLE);
+            }
+            FillWindowPixelBuffer(window, 0xFF);
+            String_Copy(bsys->msgBuffer, msg);
+            index = AddTextPrinterParameterized(window, 1, bsys->msgBuffer, 0, 0, delay, ov12_0223CF14);
+            String_Delete(msg);
+        } else {
+            MsgData *data;
+            String *msg;
+            int stringId;
+            u32 msgId;
+            int i;
+    
+            if (a2 == 0x64) {
+                stringId = trainerId * 3 + 1;
+            } else {
+                stringId = trainerId * 3 + 2;
+            }
+            
+            for (i = 0; i < 4; i++) {
+                if (PlayerProfile_GetVersion(bsys->playerProfile[i]) == 0) {
+                    break;
+                }
+            }
+    
+            if (i == 4) {
+                msgId = msg_0197_00724;
+            } else {
+                msgId = msg_0197_00723;
+            }
+    
+            data = NewMsgDataFromNarc(MSGDATA_LOAD_DIRECT, NARC_msgdata_msg, msgId, HEAP_ID_BATTLE);
+            msg = NewString_ReadMsgData(data, stringId);
+            FillWindowPixelBuffer(window, 0xFF);
+            String_Copy(bsys->msgBuffer, msg);
+            index = AddTextPrinterParameterized(window, 1, bsys->msgBuffer, 0, 0, delay, ov12_0223CF14);
+            String_Delete(msg);
+            DestroyMsgData(data);
+        }
+    } else {
+        GetTrainerMessageByIdPair(trainerId, a2, bsys->msgBuffer, HEAP_ID_BATTLE);
```

Comment:
document a2

Reply:
the function commented on implies it's a message id but I'm not even sure if that's how it's being used, since earlier it checks if it's equal to 0x64 (100) and loads strings based on that.

Reply:
fair, that's ok if it's not known then